src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -47,27 +47,22 @@ internal abstract class AbstractMakeMethodAsynchronousCodeFixProvider : CodeFixP
             }
 
             var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken) as IMethodSymbol;
+            var entryPoint = compilation.GetEntryPoint(cancellationToken);
+            bool isEntryPoint = entryPoint?.Equals(symbol) == true;
```

Comment:
why not remove hte "as IMethodSybol" and then just do:  ```var isEntryPoint = symbol.Equals(entryPoint)```?

Also, what's the perf of "GetEntryPoint"?   wouldn't it have to scan the entire compilation?  Should we instead compute this during the fix itself?

Reply:
Note: we already wrote a function to find symbols efficiently by name in the compilation by using: 

```GetSymbolsWithName```.  Unfortunately, because it takes a predicate and not a string, it needs to realize a type to get all the member symbols.  That code could be augmented to just take a string, in which case it could efficiently check if it even needed to check the members of a type or not.